### PR TITLE
migrate reference column to jsonb [no ticket; risk: low]

### PR DESCRIPTION
**Pull request by davidangb**
tested locally, and this properly migrates pre-existing data in the table (in my tests)

jsonb generally supports more functionality and could be useful for us in the future, allowing us to query against individual parts of the embedded json.

### CODE REVIEW

File: src/main/resources/db/changesets/20201204_reference_jsonb.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Add an index on workspace_id and reference_id as they're frequently looked up together
```
Comment (zloery) [inline src/main/resources/db/changesets/20201204_reference_jsonb.yaml:1]: nit: update comment

**Review by zloery — APPROVED** [review]
LGTM, one comment to change

**Review by wchamber — COMMENTED** [review]
There's a few other `json` columns in the initial schema. Can we change all of them at once? I don't think there's much downside.

File: src/main/resources/db/changesets/20201204_reference_jsonb.yaml (hunk 1 above)
Comment (davidangb) [reply src/main/resources/db/changesets/20201204_reference_jsonb.yaml]: lol thank you, good catch 🤷

**Comment by davidangb** [comment]
@wchamber I think I got them all, how's this?

**Review by wchamber — APPROVED** [review]

**Review by zloery — APPROVED** [review]

**Pull request merged**